mautic/mautic
# Fixing bug in recent activity widget

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This bug solves the issue https://github.com/mautic/mautic/issues/14326 and makes sure then when a system activity is displayed in the recent activity dashboard widget, no link to an non-existant user is added.






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a landing page and open it in an incognito tab
3. Create a contact with you current user
4. Wait till you dashboard cache runs out. You can reduce the time that widget data is cached via Configuration -> System Settings -> Cached data timeout. However keep in mind, that there is currently a bug, so setting the cached data timeout to 0 (disabling cache) doesn't work.
5. See that in the recent activity dashboard widget a contact for the non-system user a link is added but for the system user there is no link added.
